### PR TITLE
[DEV-1333] Fix sign-in and confirm sign-in UI and UX

### PR DESCRIPTION
**Pull request by marcobottaro**
Fix sign-in and confirm sign-in UI and UX.
This PR is the first result of the division of the previous PR #486 

#### List of Changes

It changes the UX of the sign-in page and the corresponding confirmation page according to Figma design

#### Screenshots (if appropriate):
Layout (take a look at the label of password's field:
image

Error management:
image

Confirmation page:
image

Confirmation page error:
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by tommaso1 — APPROVED** [review]

File: apps/nextjs-website/src/components/organisms/Auth/ConfirmLogin.tsx [hunk 1]
```diff
@@ -57,53 +64,65 @@ const ConfirmLogin = ({ email, onConfirmLogin }: confirmLoginProps) => {
...
             <Stack spacing={2} mb={4}>
               <TextField
                 variant='outlined'
                 size='small'
                 onChange={(e) => setCode(e.target.value)}
+                helperText={
+                  codeError
+                    ? confirmLogin('invalidCode')
+                    : emptyCode
+                    ? confirmLogin('emptyCode')
```
Comment (datalek) [inline apps/nextjs-website/src/components/organisms/Auth/ConfirmLogin.tsx:94]: I think the `emptyCode` key is missing:

Screenshot 2024-02-09 at 12 07 14

**Review by datalek — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/components/organisms/Auth/ConfirmLogin.tsx (hunk 1 above)
Comment (marcobottaro) [reply apps/nextjs-website/src/components/organisms/Auth/ConfirmLogin.tsx:94]: Thanks!

File: apps/nextjs-website/src/messages/it.json [hunk 2]
```diff
@@ -218,9 +223,9 @@
             "title": "Recupera password",
             "body": "Inserisci il tuo indirizzo e-mail e ti invieremo le istruzioni per impostare una nuova password.",
             "goBackToLogin": "Torna al login",
-            "send": "Invia",
+            "send": "Conferma",
             "checkEmailTitle": "Controlla l'email",
-            "checkEmail": "Se esiste un account associato a {email}, riceverai una e-mail con un link per impostare una nuova password.",
+            "checkEmail": "Se esiste un account associato a <strong>{email}</strong>, riceverai una email con un link per impostare una nuova password.",
```
[the remark this answers is not in the document]
Comment (marcobottaro) [reply apps/nextjs-website/src/messages/it.json:232]: This PR  - which is the first of five derived from #486 - also contains translations that will be used in the other pull requests derived from #486 and depending on this.
This key will be replaced here: https://github.com/pagopa/developer-portal/pull/605/files#diff-9694610a95d36ccda80762bd47476caf3b54f3c8add8fbdbc5f3fe0c19d9f936

File: apps/nextjs-website/src/messages/it.json [hunk 3]
```diff
@@ -157,10 +158,12 @@
         "signUp": {
             "action": "Iscriviti",
             "createYourAccount": "Crea il tuo account",
-            "confirmComunications": "Inviami e-mail relative alle risorse e agli aggiornamenti sui prodotti. Se questa casella è selezionata, PagoPA ti invierà di tanto in tanto delle e-mail utili e pertinenti. Puoi annullare l'iscrizione in qualsiasi momento.",
-            "acceptPolicy": "Cliccando su “Iscriviti” accetti la nostra informativa sul trattamento dei dati personali per la Privacy Policy.",
+            "passwordMismatchError": "Le password non combaciano",
+            "confirmComunications": "Inviami email relative alle risorse e agli aggiornamenti sui prodotti. Se questa casella è selezionata, PagoPA ti invierà di tanto in tanto delle email utili e pertinenti. Puoi annullare l'iscrizione in qualsiasi momento.",
+            "acceptPolicy": "Cliccando su iscriviti dichiari di aver preso visione della nostra <policy>informativa privacy</policy> e dei nostri <terms>termini e condizioni d'uso</terms>.",
```
[the remark this answers is not in the document]
Comment (marcobottaro) [reply apps/nextjs-website/src/messages/it.json:167]: Same as [here](https://github.com/pagopa/developer-portal/pull/601/files#r1487954890):
see https://github.com/pagopa/developer-portal/pull/604/files#diff-2972bcfe37c1403c5348b8df5c0c6a0d84b2e98d9d05286b5802e36a8bebe37c

File: apps/nextjs-website/src/components/organisms/Auth/ConfirmLogin.tsx [hunk 4]
```diff
@@ -1,42 +1,49 @@
...
 
   const onConfirmLoginHandler = useCallback(() => {
+    setCodeError(false);
+    setEmptyCode(false);
     setSubmitting(true);
+
     onConfirmLogin(code).catch((e) => {
-      setError(e.message);
-      setSubmitting(false);
+      if (e.message === 'Challenge response cannot be empty') {
```
[the remark this answers is not in the document]
Comment (marcobottaro) [reply apps/nextjs-website/src/components/organisms/Auth/ConfirmLogin.tsx:42]: The error `name` is the same in both cases, using the message is the only way to distinguish them.
As you can see, the design team introduced the empty code error on [Figma](https://www.figma.com/file/AhMGjqrjE10DhqyI83gZoL/23-Q3-e-Q4---DevPortal?type=design&node-id=1218-36770&mode=design&t=vd2Pks0LMDUAhvFf-4).

**Review by datalek — APPROVED** [review]

**Pull request merged**